ldi-core/ldes-client/tree-node-relations-fetcher/src/main/java/ldes/client/treenoderelationsfetcher/LdesStructureDiscoverer.java

```diff
@@ -0,0 +1,62 @@
+package ldes.client.treenoderelationsfetcher;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldi.requestexecutor.executor.RequestExecutor;
+import ldes.client.treenoderelationsfetcher.domain.valueobjects.LdesStructure;
+import ldes.client.treenoderelationsfetcher.domain.valueobjects.TreeNodeRequest;
+import ldes.client.treenoderelationsfetcher.domain.valueobjects.TreeRelation;
+import org.apache.jena.riot.Lang;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class LdesStructureDiscoverer {
+	private static final Logger log = LoggerFactory.getLogger(LdesStructureDiscoverer.class);
+	private final TreeRelationsFetcher treeNodeFetcher;
+	private final Lang sourceFormat;
+	private final RequestExecutor requestExecutor;
+	private final LdesStructure ldesStructure;
+
+	public LdesStructureDiscoverer(String startingUrl, Lang sourceFormat, RequestExecutor requestExecutor) {
+		this.sourceFormat = sourceFormat;
+		this.requestExecutor = requestExecutor;
+		this.treeNodeFetcher = new TreeRelationsFetcher(requestExecutor);
+		this.ldesStructure = new LdesStructure(startingUrl);
+	}
+
+	public LdesStructure discoverLdesStructure() {
+		getStartingTreeRelations()
+				.parallelStream()
+				.forEach(relation -> {
+					ldesStructure.addRelation(relation);
+					fetchRelations(relation);
+				});
+		return ldesStructure;
```

Comment:
Functionally, this code is working, so i'll leave this as a comment rather than an action point to change anything:
I would expect this method to stand on its own. You do your calls to getStartingRelations (which return void) and suddenly there is a finished ldes structure..

What I would expect is that instead of a forEach, this would be a map and the final product is returned